test/e2e/windows/gmsa_full.go

```diff
@@ -0,0 +1,416 @@
+			serviceAccountName := createServiceAccount(f)
+
+			ginkgo.By("binding the RBAC role to the service account")
+			bindRBACRoleToServiceAccount(f, serviceAccountName, rbacRoleName)
+
+			ginkgo.By("creating a pod using the GMSA cred spec")
+			podName := createPodWithGmsa(f, serviceAccountName)
+
+			// nltest /QUERY will only return successfully if there is a GMSA
+			// identity configured, _and_ it succeeds in contacting the AD controller
+			// and authenticating with it.
+			ginkgo.By("checking that nltest /QUERY returns successfully")
+			var output string
+			gomega.Eventually(func() bool {
+				output, err = runKubectlExecInNamespace(f.Namespace.Name, podName, "nltest", "/QUERY")
+				return err == nil
+			}, 1*time.Minute, 1*time.Second).Should(gomega.BeTrue())
+
+			expectedSubstr := "The command completed successfully"
+			if !strings.Contains(output, expectedSubstr) {
+				e2elog.Failf("Expected %q to contain %q", output, expectedSubstr)
+			}
+		})
+	})
+})
+
+// findPreconfiguredGmsaNode finds node with the gmsaFullNodeLabel label on it.
+func findPreconfiguredGmsaNodes(c clientset.Interface) []v1.Node {
+	nodeOpts := metav1.ListOptions{
+		LabelSelector: gmsaFullNodeLabel,
+	}
+	nodes, err := c.CoreV1().Nodes().List(nodeOpts)
+	if err != nil {
+		e2elog.Failf("Unable to list nodes: %v", err)
+	}
+	return nodes.Items
+}
+
+// retrieveCRDManifestFileContents retrieves the contents of the file
+// at gmsaCrdManifestPath on node; it does so by scheduling a single pod
+// on nodes with the gmsaFullNodeLabel label with that file's directory
+// mounted on it, and then exec-ing into that pod to retrieve the file's
+// contents.
+func retrieveCRDManifestFileContents(f *framework.Framework, node v1.Node) string {
+	podName := "retrieve-gmsa-crd-contents"
+	// we can't use filepath.Dir here since the test itself runs on a Linux machine
+	splitPath := strings.Split(gmsaCrdManifestPath, `\`)
+	dirPath := strings.Join(splitPath[:len(splitPath)-1], `\`)
+	volumeName := "retrieve-gmsa-crd-contents-volume"
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: f.Namespace.Name,
+		},
+		Spec: v1.PodSpec{
+			NodeSelector: node.Labels,
+			Containers: []v1.Container{
+				{
+					Name:  podName,
+					Image: imageutils.GetPauseImageName(),
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      volumeName,
+							MountPath: dirPath,
+						},
+					},
+				},
+			},
+			Volumes: []v1.Volume{
+				{
+					Name: volumeName,
+					VolumeSource: v1.VolumeSource{
+						HostPath: &v1.HostPathVolumeSource{
+							Path: dirPath,
+						},
+					},
+				},
+			},
+		},
+	}
+	f.PodClient().CreateSync(pod)
+
+	// using powershell and using forward slashes avoids the nightmare of having to properly
+	// escape quotes and backward slashes
+	output, err := runKubectlExecInNamespace(f.Namespace.Name, podName, "powershell", "type", strings.ReplaceAll(gmsaCrdManifestPath, `\`, "/"))
```

Comment:
could go with powershell get-content here. Type is aliased to get-content, but much less known.